enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/SlottedRewriter.scala

```diff
@@ -146,8 +146,10 @@ class SlottedRewriter(tokenContext: TokenContext) {
       case e@Equals(Variable(k1), Variable(k2)) => // TODO: Handle nullability
         val slot1 = pipelineInformation(k1)
         val slot2 = pipelineInformation(k2)
-        if (slot1.typ == slot2.typ)
+        if (slot1.typ == slot2.typ && PipelineInformation.isLongSlot(slot1)) {
```

Comment:
Shouldn't we check both slots here and not just have an assertion, can't we get a Reference Node and a Slot Node here?

Reply:
If they have the same type I assumed they should have the same slot type or something is wrong with slot allocation.